src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -124,7 +131,8 @@ public sealed override object ReadInline(object item, IEdmTypeReference edmType,
                 throw Error.ArgumentNull("readContext");
             }
 
-            if (!String.IsNullOrEmpty(resourceWrapper.Resource.TypeName) && structuredType.FullName() != resourceWrapper.Resource.TypeName)
+
+            if (!String.IsNullOrEmpty(resourceWrapper.ResourceBase.TypeName) && structuredType.FullName() != resourceWrapper.ResourceBase.TypeName)
```

Comment:
```suggestion
            if (!string.IsNullOrEmpty(resourceWrapper.ResourceBase.TypeName) && structuredType.FullName() != resourceWrapper.ResourceBase.TypeName)
```
nit:

Reply:
I think string - lowercase `s` is the conventional C# style.